src/rtFileDownloader.cpp

```diff
@@ -783,7 +806,7 @@ bool rtFileDownloader::checkAndDownloadFromCache(rtFileDownloadRequest* download
   rtData data;
   if ((NULL != rtFileCache::instance()) && (RT_OK == rtFileCache::instance()->httpCacheData(downloadRequest->fileUrl(),cachedData)))
   {
-    err = cachedData.data(data);
+    err = cachedData.data(data, downloadRequest->readCachedDataInChunksEnabled());
```

Comment:
We should avoid this data() call if deferCacheRead() is true.  Instead, please call an API on rtHttpCacheData that will setup rtHttpCache to be read lazily.